R/extract_string.R

```diff
@@ -1,3 +1,6 @@
+
```

Comment:
This could do with full roxygen documentation even if not being exported ie exclude `@export`

Reply:
@stephlocke should this be the case for all function documents that don't currently have them? 

How do I decide which ones should include/exclude @export? Is there a rule?

Reply:
export is only used for user facing functions. In this case, if the function does not have any roxygen docs associated with it, then it does not need an export. Additionally, I would include a `@noRd` tag so that the documentation doesn't get built into. Here's an example of documentation of an internal function for the incidence package: https://github.com/reconhub/incidence/blob/51269f1e0e8cfe10181b3940745112a0f84eadf9/R/check_dots.R#L1-L13

Reply:
Thanks Zhian that is helpful! I'm on it today.

Reply:
General rule of thumb I apply for whether to export something is whether it aligns to the core goal of the package. If it's written to help make my life easier as the developer of functions, reduce my code per function etc, then it shouldn't get exported. The more functionality exported, the more that needs supporting around breaking changes so I prefer not to export a function unless it's part of the core package interface.